### PR TITLE
NAR local configuration

### PR DESCRIPTION
**Pull request by PtrMan**
Allows to run multiple NAR's with different parameters

### CODE REVIEW

**Comment by PtrMan** [comment]
Exception in thread "main" java.lang.NoSuchFieldError: decisionThreshold
	at org.opennars.gui.NARControls.newParameterPanel(NARControls.java:742)
	at org.opennars.gui.NARControls.<init>(NARControls.java:329)
	at org.opennars.gui.NARSwing.<init>(NARSwing.java:149)
	at org.opennars.gui.NARSwing.<init>(NARSwing.java:141)
	at org.opennars.gui.NARSwing.main(NARSwing.java:196)

**Comment by PtrMan** [comment]
hm the build shouldn't pass

**Comment by patham9** [comment]
Why can't the parameters be part of Memory, why pass it as parameters in all these methods?

**Comment by PtrMan** [comment]
because it should be a part of nar

**Comment by PtrMan** [comment]
need to invest a few braincycles more at this

**Comment by PtrMan** [comment]
ok fixed the code issues

File: src/main/java/org/opennars/control/DerivationContext.java [hunk 1]
```diff
@@ -43,10 +44,13 @@
     protected Sentence currentBelief;
     protected Stamp newStamp;
     public StampBuilder newStampBuilder;
+
+    public NarParameters narParameters;
     
-    public DerivationContext(final Memory mem) {
+    public DerivationContext(final Memory mem, NarParameters narParameters) {
```
Comment (freemo) [inline src/main/java/org/opennars/control/DerivationContext.java]: parameter can be final

File: src/main/java/org/opennars/control/GeneralInferenceControl.java [hunk 2]
```diff
@@ -25,12 +25,13 @@
...
 import org.opennars.io.events.Events;
 import org.opennars.main.Parameters;
+import org.opennars.main.NarParameters;
 import org.opennars.storage.Memory;
 
 /** Concept reasoning context - a concept is "fired" or activated by applying the reasoner */
 public class GeneralInferenceControl {
     
-    public static void selectConceptForInference(final Memory mem) {
+    public static void selectConceptForInference(final Memory mem, NarParameters narParameters) {
```
Comment (freemo) [inline src/main/java/org/opennars/control/GeneralInferenceControl.java]: can be final

File: src/main/java/org/opennars/entity/Concept.java [hunk 3]
```diff
@@ -542,7 +543,7 @@ public void end() {
      * @param time The current time
      * @return The selected TermLink
      */
-    public TermLink selectTermLink(final TaskLink taskLink, final long time) {
+    public TermLink selectTermLink(final TaskLink taskLink, final long time, NarParameters narParameters) {
```
Comment (freemo) [inline src/main/java/org/opennars/entity/Concept.java]: can be final

File: src/main/java/org/opennars/entity/TaskLink.java [hunk 4]
```diff
@@ -147,10 +148,11 @@ public final short getIndex(final int i) {
      * @param currentTime The current time
      * @return Whether they are novel to each other
      */
-    public boolean novel(final TermLink termLink, final long currentTime) {
-        return novel(termLink, currentTime, false);
+    public boolean novel(final TermLink termLink, final long currentTime, NarParameters narParameters) {
```
Comment (freemo) [inline src/main/java/org/opennars/entity/TaskLink.java]: can be final

File: src/main/java/org/opennars/entity/TaskLink.java [hunk 5]
```diff
@@ -147,10 +148,11 @@ public final short getIndex(final int i) {
...
+        return novel(termLink, currentTime, narParameters, false);
     }
-    public boolean novel(final TermLink termLink, final long currentTime, final boolean transformTask) {
+
+    public boolean novel(final TermLink termLink, final long currentTime, NarParameters narParameters, final boolean transformTask) {
```
Comment (freemo) [inline src/main/java/org/opennars/entity/TaskLink.java]: parameter can be final

**Pull request merged**